Added 'Blessed' icons for core plugins

hrithiksawhney opened the pull request:
## Hey, I just made a Pull Request!

In this Pull Request, I have selected a few icons from v4 MUI Icons set to represent major parts of Backstage, for use in buttons and links into these components: 
catalog: MenuBook Icon
scaffolder & templates: CreateNewFolder Icon
techdocs: Subject Icon
search: Search Icon

I have added these icons to the list.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

freben commented:
Alright, two more things.

You'll need to add a changeset, and add a DCO signature onto your commit. See the details of the failed build entry here for details, and some notes in CONTRIBUTING.md

hrithiksawhney commented:
> Alright, two more things.
> 
> You'll need to add a changeset, and add a DCO signature onto your commit. See the details of the failed build entry here for details, and some notes in CONTRIBUTING.md

Hi, I have added the changeset and also added a DCO signature onto my commit.

freben commented on .changeset/spicy-panthers-thank.md:
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/core-app-api': major
```
Not a major change - leave this as `patch`

hrithiksawhney replied:
Done

freben reviewed (approved):
Alright!

freben commented:
Just have to wait for it to turn green and then I'll merge some time later today

freben commented:
You'll notice it errors on unknown words in the changeset. You should probably just simplify it to say "Added default icons for the scaffolder, techdocs, and search."

hrithiksawhney commented:
> You'll notice it errors on unknown words in the changeset. You should probably just simplify it to say "Added default icons for the scaffolder, techdocs, and search."

Yes, I have made changes to changeset.
